recipes/conversation_assistant/conversation.js

```diff
@@ -111,16 +169,16 @@ while (true) {
             case 'shine':
                 {
                     let misunderstood = false;
-                    if (response.object.entities !== undefined) {
-                        const entity = response.object.entities[0];
-                        if (entity !== undefined && entity.value !== undefined) {
-                            const color = entity.value;
-                            await tj.speak(response.description);
-                            tj.shine(color);
-                            spoken = true;
-                        } else {
-                            misunderstood = true;
-                        }
+
+                    // colors to detect from the user utterance
+                    const regex = /(aqua|red|green|white|blue|orange|yellow|violet|pink|on|off)/g;
```

Comment:
tjbotlib defines a list of colors… I think you can just pass the string color to `normalizeColor()` and magic will happen to give you a color back.

Reply:
Right - that's what this is attempting to do. It is pulling the color out of the user utterance (if a user says "shine the light to violet") then it pulls out "violet" and passes that to `tj.shine()`